New Approach/HDD/NewRegionsMCCMRG-Deltas/Functions.R

```diff
@@ -110,15 +111,20 @@ if (CalclulateDelta){
       Deltas <- vector(mode = "list", length = nrow(MCC_MRG_Grid))
       names(Deltas) <- as.character(Deltas_names[["MCC_MRG"]])
 
-      ## Filtering for regions loonger than 3bp-long has been done on line 80
+      ## Filtering for regions loonger than 3bp-long has been done on line 80 (RegionMat_MCC_MRG)
       LaidPairs <- future_map(seq_len(length(NonOverlappedExons)),
         ~findOverlapPairs(granges(RegionMat_MCC_MRG[[.x]][['regions']]),
           granges(NonOverlappedExons[[.x]])), .id = "Chr")
       ### Filtering Er's laid on multiple Exons
-      Ers_without_multi_Exons_Index <- future_map(LaidPairs,~ unite(data.frame(.x@first), col = "Region", sep="-", remove = TRUE]) %>%
-       table() %>% as_tibble() %>% dplyr::filter(n == 1) %>% dplyr::separate(col = "Region",
-        into = c("seqnames", "start", "end", "width", "strand"),sep = "-", remove = TRUE) %>% see####################
-      LaidPairs <- future_map2(LaidPairs, Ers_without_multi_Exons_Index, ~ dplyr::semi_join(x = ..1, y = ..2))
+      Ers_Laid_One_Exon <- future_map(LaidPairs,~ unite(data.frame(.x@first), col = "Region", sep="-", remove = TRUE) %>% table() %>% as_tibble() %>%
+          dplyr::filter(n == 1) %>% tidyr::separate(col = ".", into = c("seqnames", "start", "end", "width", "strand"),sep = "-", remove = TRUE) %>% dplyr::select(-c(n)) )
+     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "seqnames", as.factor(.x[[1]]) ))
+     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "start", as.integer(.x[["start"]]) ))
+     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "end", as.integer(.x[["end"]]) ))
+     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "width", as.integer(.x[["width"]]) ))
+     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "strand", as.factor(.x[["strand"]]) ))
```

Comment:
I was making a mistake in comparison. LaidPairs is actually a pair list object returning some ERs laid multiple time on exons for each laying make the ERs appear more than once whereas I was subtracting number of Ers_Laid_One_Exon from total number of LaidPairs@first and then checking if the outcome is equal to number of Ers_Laid_Multiple_Exon. The preceding subtraction and comparison is false due to the fact that Ers_Laid_Multiple_Exon is retrieved the same as Ers_Laid_One_Exon and so that it doesn't include times the Er has been duplicated in LaidPairs. Therefore, the correct comparison would be comparing  number of unique names of   ER's inLaidParis@first as the source for subtraction and addition of either Ers_Laid_One_Exon or Ers_Laid_Multiple_Exon
```suggestion
     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "strand", as.factor(.x[["strand"]]) ))
```
```suggestion
     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "strand", as.factor(.x[["strand"]]) ))
```
```suggestion
     Ers_Laid_One_Exon <- map(Ers_Laid_One_Exon, ~ base::`$<-`(.x, "strand", as.factor(.x[["strand"]]) ))
```